api-put-bucket.go

```diff
@@ -313,6 +332,89 @@ func (c Client) removeBucketLifecycle(ctx context.Context, bucketName string) er
 	return nil
 }
 
+// SetBucketEncryption sets the default encryption configuration on an existing bucket.
+func (c Client) SetBucketEncryption(bucketName string, configuration ServerSideEncryptionConfiguration) error {
+	return c.SetBucketEncryptionWithContext(context.Background(), bucketName, configuration)
+}
+
+// SetBucketEncryptionWithContext sets the default encryption configuration on an existing bucket with a context to control cancellations and timeouts.
+func (c Client) SetBucketEncryptionWithContext(ctx context.Context, bucketName string, configuration ServerSideEncryptionConfiguration) error {
+	// Input validation.
+	if err := s3utils.CheckValidBucketName(bucketName); err != nil {
+		return err
+	}
+
+	xmlBytes, err := xml.Marshal(&configuration)
```

Comment:
This whole thing can be rewritten as following 

```go
        buf, err := xml.Marshal(&configuration)
        if err != nil {
                return err
        }

        // Get resources properly escaped and lined up before
        // using them in http request.
        urlValues := make(url.Values)
        urlValues.Set("encryption", "")

        reqMetadata := requestMetadata{
                bucketName:       bucketName,
                queryValues:      urlValues,
                contentBody:      bytes.NewReader(buf),
                contentLength:    int64(len(buf)),
                contentMD5Base64: sumMD5Base64(buf),
        }
```

Reply:
ok